src/main/java/org/apache/mesos/offer/ExecutorRequirement.java

```diff
@@ -56,7 +57,9 @@ private ExecutorRequirement(ExecutorInfo executorInfo)
         validateExecutorInfo(executorInfo);
         this.executorInfo = executorInfo;
         this.resourceRequirements =
-                RequirementUtils.getResourceRequirements(this.executorInfo.getResourcesList());
+                RequirementUtils.getResourceRequirements(executorInfo.getResourcesList());
+        this.dynamicPortRequirements =
```

Comment:
Any reason for the separate list given that DynamicPortRequirement extends ResourceRequirement? if so, maybe add a comment describing why